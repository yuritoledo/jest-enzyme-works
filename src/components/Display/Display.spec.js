import React from 'react'
import { shallow } from 'enzyme'
import Display from './Display'

describe('Display', () => {
	let wrapper = shallow(<Display displayValue={''} />)

	beforeEach(() => {
		wrapper = shallow(<Display displayValue={''} />)
	})

	test('should render correctly', () => {
		expect(wrapper).toMatchSnapshot()
	})


	test('should render a div', () => {
		expect(wrapper.find('div')).toBeTruthy()
	})

	test('should render the value of displayValueProp', () => {
		wrapper.setProps({ displayValue: 'test' })
		expect(wrapper.text()).toEqual('test')
	})



})
