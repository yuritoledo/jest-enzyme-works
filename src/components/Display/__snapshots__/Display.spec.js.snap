// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Display
    displayValue=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className="display-value"
/>,
      "className": "display-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "className": "display-value",
      },
      "ref": null,
      "rendered": "",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="display-value"
/>,
        "className": "display-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "display-value",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
